okreplay-tests/src/test/java/okreplay/AnnotationTest.kt

```diff
@@ -0,0 +1,105 @@
+package okreplay
+
+import com.google.common.io.Files
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.codehaus.groovy.runtime.ResourceGroovyMethods
+import org.junit.AfterClass
+import org.junit.Assert
+import org.junit.BeforeClass
+import org.junit.FixMethodOrder
+import org.junit.Rule
+import org.junit.Test
+import java.io.File
+import java.io.IOException
+import java.net.HttpURLConnection
+
+@FixMethodOrder // The test methods are dependent on one another
+class AnnotationTest {
+    var configuration: OkReplayConfig = OkReplayConfig.Builder()
+        .tapeRoot(tapeRoot)
+        .defaultMode(TapeMode.READ_WRITE)
+        .interceptor(OkReplayInterceptor())
+        .build()
+
+    @Rule
+    @JvmField
+    var recorder = RecorderRule(configuration)
+
+    var client: OkHttpClient = OkHttpClient.Builder()
+        .addInterceptor(configuration.interceptor())
+        .build()
+
+    @Test
+    fun noTapeIsInsertedIfThereIsNoAnnotationOnTheTest() {
+        Assert.assertNull(recorder.tape)
+    }
+
+    @Test
+    @OkReplay(tape = "annotation_test", mode = TapeMode.READ_WRITE)
+    fun annotationOnTestCausesTapeToBeInserted() {
+        Assert.assertEquals("annotation_test", recorder.tape!!.name)
+    }
+
+    @Test
+    fun tapeIsEjectedAfterAnnotatedTestCompletes() {
+        Assert.assertNull(recorder.tape)
+    }
+
+    @Test
+    @OkReplay(tape = "annotation_test", mode = TapeMode.READ_WRITE)
+    @Throws(IOException::class)
+    fun annotatedTestCanRecord() {
+        endpoint.enqueue(MockResponse().setBody("Echo"))
+        val request: Request = Request.Builder().url(endpoint.url("/")).build()
+        val response = client.newCall(request).execute()
+        Assert.assertEquals(HttpURLConnection.HTTP_OK.toLong(), response.code.toLong())
+        Assert.assertEquals("OkReplay", response.header(Util.VIA))
+        Assert.assertEquals("REC", response.header(Headers.X_OKREPLAY))
+    }
+
+    @Test
+    @OkReplay(tape = "annotation_test", mode = TapeMode.READ_WRITE)
+    @Throws(IOException::class)
+    fun annotatedTestCanPlayBack() {
+        endpoint.enqueue(MockResponse().setBody("Echo"))
+        val request: Request = Request.Builder().url(endpoint.url("/")).build()
+        val response = client.newCall(request).execute()
+        Assert.assertEquals(HttpURLConnection.HTTP_OK.toLong(), response.code.toLong())
+        Assert.assertEquals("OkReplay", response.header(Util.VIA))
```

Comment:
Some of the test strings in this class could be extracted to a companion object as well

Reply:
Might as well ... I don't think we're going to preserve any history with the Spock test.